lightning/src/ln/peer_handler.rs

```diff
@@ -1088,15 +1095,107 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 			}
 		};
 	}
+
+// return a set of descriptors for all the peers in our peer manager
+fn get_des(&self) -> Vec<Descriptor> {
+	let mut des_set: Vec<Descriptor> = Vec::new();
+	let mut peers_lock = self.peers.lock().unwrap();
+	let peers = peers_lock.borrow_parts();
+			
+	for (Descriptor, mut Peer) in peers.peers.iter_mut() {
+
+				//read all events that have been sent from the Peer
+				let mut descriptor = Descriptor.clone();
+				des_set.push(descriptor);
+	}
+	des_set
+
+}
+
+//take a vector of descriptors and performs do_read_event on each of them
+fn mass_do_read_event(&self, mut des_set: Vec<Descriptor>)
+{
+	for (mut Descriptor) in des_set.iter_mut(){
+		let data: Vec<u8> = Vec::new();
+				
+		let res = match self.do_read_event(&mut Descriptor, data){
+					Ok(pause_read) => pause_read,
+					Err(e) => panic!("something is wrong"),
+		};
+	}
+
+}
+
+// iterate through our peers, if there are outstanding pings add a descriptor for the peer to a vector to be removed
+//otherwise ping all peers
+fn handle_pings(&self)-> Vec<Descriptor>{
+
+			let mut des_set: Vec<Descriptor> = Vec::new();
+			let mut peers_lock = self.peers.lock().unwrap();
+			let peers = peers_lock.borrow_parts();
+			
+			for (Descriptor, mut Peer) in peers.peers.iter_mut() {
+
+				// Disconect the Peer if there is an outstanding ping for which we have not been ponged
+				if Peer.ping_tracker > 0 {
+					let mut descriptor = Descriptor.clone();
+					des_set.push(descriptor);
+				}
+
+			else {
+		
+			let ping = msgs::Ping {
+ 				ponglen: 64,
```

Comment:
Why 64? 0 should be no different, no?

Reply:
So it appears in msgs.rs the test function encoding_ping() uses 64 for both the ponglen and the byteslen (line 2044) which is why I originally picked it. 
From what i understood  the only stipulation is that the ping messages ponglen is less than 65532 otherwise it fails so is 65531 what we wanted?

I checked out the lighting BOLTs as well as  took a cursory look at c-lightning for possible insight and it appears that the specific length is not specificed? (it just has to be under 65532).

Reply:
"If `num_pong_bytes` is less than 65532, MUST respond by sending a `pong` message with `byteslen` equal to `num_pong_bytes`

Reply:
cool, yeah I think I will just go with 65531 unless there is some convention to use a specific number.

Reply:
Anything over 0 has no effect on actual operation (sadly there is no real nonce here that we can set), and wastes bandwidth for no change. We should just stick with 0 always.

Reply:
ah okay that is odd lol. I was a little confused by your original post on the matter.

Reply:
You don't *need* to use anything, its just wasteful to use > 0.